File: Programmin Fundamentals/Homeworks and Labs/DictionariesLambdaAndLINQ-Exercises/02.PhonebookUpgrade/PhonebookUpgrade.cs
using System;
using System.Collections.Generic;
using System.Linq;

public class PhonebookUpgrade
{
    public static void Main()
    {
        string inputLine = Console.ReadLine();

        var phonebook = new SortedDictionary<string, string>();

        while (!inputLine.Equals("END"))
        {
            string[] comandArgs = inputLine.Split().ToArray();

            if (comandArgs[0].Equals("A"))
            {
                string name = comandArgs[1];
                string phoneNumber = comandArgs[2];

                if (!phonebook.ContainsKey(name))
                {
                    phonebook.Add(name, string.Empty);
                }

                phonebook[name] = phoneNumber;
            }
            else if (comandArgs[0].Equals("S"))
            {
                string name = comandArgs[1];

                if (phonebook.ContainsKey(name))
                {
                    Console.WriteLine($"{name} -> {phonebook[name]}");
                }
                else
                {
                    Console.WriteLine($"Contact {name} does not exist.");
                }
            }
            else
            {
                foreach (var contact in phonebook)
                {
                    Console.WriteLine($"{contact.Key} -> {contact.Value}");
                }
            }

            inputLine = Console.ReadLine();
        }
    }
}

File: Software Technologies/Java Basics - Exercises/src/PhonebookUpgrade.java
import java.util.HashMap;
import java.util.Scanner;
import java.util.TreeMap;

public class PhonebookUpgrade {
    public static void main(String [] args)
    {
        Scanner scanner = new Scanner(System.in);
        String inputLine = scanner.nextLine();

        TreeMap<String,String> phonebook = new TreeMap<String,String>();

        while (!inputLine.equals("END")) {
            String[] comandArgs = inputLine.split("\\s");

            if (comandArgs[0].equals("A")) {
                String name = comandArgs[1];
                String phoneNumber = comandArgs[2];

                if (!phonebook.containsKey(name)) {
                    phonebook.put(name, "");
                }

                phonebook.put(name, phoneNumber);

            }
            else if (comandArgs[0].equals("S")) {
                String name = comandArgs[1];

                if (phonebook.containsKey(name)) {
                    System.out.println(name+" -> "+phonebook.get(name));
                }
                else {
                    System.out.println("Contact "+name+" does not exist.");
                }
            }
            else
            {
                for (String key : phonebook.keySet())
                {
                    System.out.println(key+" -> "+phonebook.get(key));
                }
            }

            inputLine = scanner.nextLine();
        }
    }
}
